rust/datafusion/src/logical_plan/builder.rs

```diff
@@ -220,20 +260,28 @@ impl LogicalPlanBuilder {
         }))
     }
 
-    /// Apply an aggregate
-    pub fn aggregate(&self, group_expr: &[Expr], aggr_expr: &[Expr]) -> Result<Self> {
-        let mut all_expr = group_expr.to_vec();
```

Comment:
it was even worse here -- *two* copies of each input expr were made -- one copy to pass to `validate_unique_names` and once to put on the `LogicalPlanAggregate`.

Admittedly, removing the second copy doesn't require changing the signature of this function